lib/engine/step/g_1848/dutch_auction.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require_relative '../base'
+require_relative '../auctioner'
+
+module Engine
+  module Step
+    module G1848
+      class DutchAuction < Base
+        include Auctioner
+
+        attr_reader :companies
+
+        # On your turn, must buy any private or reduce price of any private
+        # Once you own a private, you may pass
+        # if everything is reduced to minimum price, and nobody has bought a private, you must buy one
+        # if any privates are left and everyone passes in a row, owned privates pay and auction continues
+        # if someone has bought a private and everything is minimum, you can pass
+
+        ACTIONS = %w[bid reduce].freeze
+        ACTIONS_WITH_PASS = %w[bid reduce pass].freeze
+
+        MIN_PRICES = {
```

Comment:
why is this here? it's not the value of the company?

Reply:
Privates in this game are a bit weird, for instance

P1 has a "value" of 40 (can be bought in by a corp for up to 40)

P1 has a "price" of 30 (can be purchased by a player for 30)

P1 has a "minimum price" of zero (can be discounted by players taking "reduce" actions for up to 30$ total)

Other privates are similar, and each one can only be discounted by 30$ total

The `discount` data member is overloaded to also handle the difference between "price" and "value", so we can't just check whether discount >= 30 and so we have this hardcoded hash